src/pages/settings/Subscription/CardSection/CardSection.tsx

```diff
@@ -9,23 +8,24 @@ import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import DateUtils from '@libs/DateUtils';
-import ONYXKEYS from '@src/ONYXKEYS';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
-import PreTrialBillingBanner from './BillingBanner/PreTrialBillingBanner';
+import SubscriptionBillingBanner from './BillingBanner/SubscriptionBillingBanner';
 import CardSectionActions from './CardSectionActions';
 import CardSectionDataEmpty from './CardSectionDataEmpty';
+import CardSectionUtils from './utils';
 
 function CardSection() {
     const {translate, preferredLocale} = useLocalize();
     const styles = useThemeStyles();
     const theme = useTheme();
-    const [fundList] = useOnyx(ONYXKEYS.FUND_LIST);
 
-    const defaultCard = useMemo(() => Object.values(fundList ?? {}).find((card) => card.isDefault), [fundList]);
+    const defaultCard = CardSectionUtils.getCardForSubscriptionBilling();
 
     const cardMonth = useMemo(() => DateUtils.getMonthNames(preferredLocale)[(defaultCard?.accountData?.cardMonth ?? 1) - 1], [defaultCard?.accountData?.cardMonth, preferredLocale]);
 
-    const BillingBanner = <PreTrialBillingBanner />;
+    const {title, subtitle, isError, icon, rightIcon} = CardSectionUtils.getBillingStatus(translate, defaultCard?.accountData?.cardNumber ?? '');
+
+    const shouldShowBanner = !!title || !!subtitle;
```

Comment:
You just removed the `<PreTrialBillingBanner />` component 😅 We need it, so I suggest having a logic like this:

```tsx
    let BillingBanner: React.ReactNode | undefined;
    if (CardSectionUtils.shouldShowPreTrialBillingBanner()) {
        BillingBanner = <PreTrialBillingBanner />;
    } else {
        // TODO: Add other billing banners here.
    }
```

Reply:
Are we going to have more banner types? Then I recommend we have a helper method returning an enum type, instead of a place if..else.